src/plots/plots.js

```diff
@@ -468,7 +477,29 @@ plots.supplyDefaults = function(gd) {
 
     // first fill in what we can of layout without looking at data
     // because fullData needs a few things from layout
-    plots.supplyLayoutGlobalDefaults(newLayout, newFullLayout);
+
+    if(oldFullLayout._initialAutoSizeIsDone) {
+        // coerce the updated layout while preserving width and height
+        var oldWidth = oldFullLayout.width,
```

Comment:
I've restored backwards-compatibility and set width and height in `gd.layout` when `layout.autosize` is false.